python/lsst/jointcal/jointcal.py

```diff
@@ -910,6 +911,9 @@ def _extract_detector_catalog_from_visit_catalog(self, table, visitCatalog, dete
 
         catalog = lsst.afw.table.SourceCatalog(table)
         matched = visitCatalog[detectorColumn] == detectorId
+        n = sum(matched)
+        if n == 0:
```

Comment:
What happens if somehow the detectorIds all end up 0?

Reply:
`matched` here is a bool array, so `sum(matched)` gives the number of elements that matched that `detectorId`. If they're all 0, then `sum(matched)` will be non-zero.